Fix initialization from incompatible pointer type compile warnings

We should fix these compile warnings:
```
/bsonjs.c -o build/temp.linux-x86_64-3.10/src/bsonjs.o
src/bsonjs.c:234:14: warning: initialization from incompatible pointer type [-Wincompatible-pointer-types]
     {"dump", dump, METH_VARARGS | METH_KEYWORDS, dump__doc__},
              ^~~~
src/bsonjs.c:234:14: note: (near initialization for ‘BsonjsClientMethods[0].ml_meth’)
src/bsonjs.c:235:15: warning: initialization from incompatible pointer type [-Wincompatible-pointer-types]
     {"dumps", dumps, METH_VARARGS | METH_KEYWORDS, dumps__doc__},
               ^~~~~
```

--- a/src/bsonjs.c
+++ b/src/bsonjs.c
@@ -231,8 +231,8 @@ loads(PyObject *self, PyObject *args)
 }
 
 static PyMethodDef BsonjsClientMethods[] = {
-    {"dump", dump, METH_VARARGS | METH_KEYWORDS, dump__doc__},
-    {"dumps", dumps, METH_VARARGS | METH_KEYWORDS, dumps__doc__},
+    {"dump", (PyCFunction)(void(*)(void))dump, METH_VARARGS | METH_KEYWORDS, dump__doc__},
+    {"dumps", (PyCFunction)(void(*)(void))dumps, METH_VARARGS | METH_KEYWORDS, dumps__doc__},
     {"load", load, METH_VARARGS, load__doc__},
     {"loads", loads, METH_VARARGS, loads__doc__},
     {NULL, NULL, 0, NULL}

